x-pack/legacy/plugins/siem/public/apps/start_app.tsx

```diff
@@ -44,7 +40,7 @@ const StartApp: FC<AppFrontendLibs> = memo(libs => {
   const store = createStore(undefined, libs$.pipe(pluck('apolloClient')));
```

Comment:
@rudolf we have a few uses of `chrome.getUiSettingsClient` remaining in functions that we use to populate part of our initial redux state (time filter preferences, etc.). Here's how we're thinking of tackling that, just wanted a sanity check from your end: 

1. Define the `setup()` method of our plugin
2. Fetch those UI settings in setup, and do some merging with the rest of our static state to create our store object
3. Pass that store to the store Provider (via a private plugin property?) during the `start()` phase

Reply:
Step 1 & 2 sound fine, however step 3 is a bit off. Registering applications is done during `setup` via the `core.application.register` API. You'll be able to use your initial store right there without using a private property on your plugin class.